app/src/main/java/com/keylesspalace/tusky/usecase/TimelineCases.kt

```diff
@@ -48,52 +48,42 @@ class TimelineCases @Inject constructor(
     private val eventHub: EventHub
 ) {
 
-    /**
-     * Unused yet but can be use for cancellation later. It's always a good idea to save
-     * Disposables.
-     */
-    private val cancelDisposable = CompositeDisposable()
-
-    fun reblog(statusId: String, reblog: Boolean): Single<Status> {
-        val call = if (reblog) {
+    suspend fun reblog(statusId: String, reblog: Boolean): NetworkResult<Status> {
```

Comment:
what's not great about all of these being `suspend` is that they will be cancelled together with the caller's scope. So we might send a network request but not apply `dispatch` because the screen is already gone

Reply:
Hah, that's been on my list of things to fix too.

There's an application level scope to use for stuff like this, [MainScope](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-main-scope.html).

Use it like this:

```
MainScope().launch {
    // do stuff that needs to be run on the UI thread or calls suspend functions
    // as MainScope uses Dispatchers.Main
}
```

(or store it in an `Application` level `val`, or similar. Then use this for network operations that should outlast the current screen.

That's still got the potential for silent failures though. I think the correct fix is https://github.com/tuskyapp/Tusky/issues/2741#issuecomment-1407430453, but that's a lot more work.

Reply:
> what's not great about all of these being suspend is that they will be cancelled together with the caller's scope. So we might send a network request but not apply dispatch because the screen is already gone

yes, but that is also the behavior before these changes